src/pages/Signup.jsx

```diff
@@ -1,21 +1,78 @@
 import { Container, Box, Typography, Stack, Paper, TextField, Button, IconButton } from "@mui/material";
 import React, { useState } from "react";
 import { Link } from "react-router-dom";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
 import { EyeOutlined, EyeInvisibleOutlined } from "@ant-design/icons";
 
 const Signup = () => {
+  const navigate = useNavigate();
+
   const [username, setUsername] = useState("");
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [showPassword, setShowPassword] = useState(false);
+  const [isDataExist, setIsDataExist] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [usernameLength, setUsernameLength] = useState(false);
+  const [isEmailEmpty, setIsEmailEmpty] = useState(false);
+  const [passwordLength, setPasswordLength] = useState(false);
 
-  const handleTogglePasswordVisibility = () => {
-    setShowPassword((prevShowPassword) => !prevShowPassword);
+  const handleSignUp = async (e) => {
+    e.preventDefault();
+
+    if (username.length < 4) {
+      setUsernameLength(true);
+      return;
+    }
+
+    if (email.length === 0) {
+      setIsEmailEmpty(true);
+      return;
+    }
+
+    if (password.length < 4) {
+      setPasswordLength(true);
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      const response = await axios.post(
+        `${import.meta.env.API_BASE_URL}/user/signup`,
```

Comment:
please move the endpoint hit to `src/api/signup.js`

you can refer to the `src/api/category.js` for example

Reply:
I moved it.